frame/merkle-mountain-range/src/mmr/mmr.rs

```diff
@@ -24,27 +24,35 @@ use crate::{
 	primitives::{self, Error, NodeIndex},
 	Config, HashingOf,
 };
-#[cfg(not(feature = "std"))]
-use sp_std::vec;
+use sp_std::prelude::*;
 
-/// Stateless verification of the leaf proof.
-pub fn verify_leaf_proof<H, L>(
+/// Stateless verification of the proof for a batch of leaves.
+/// Note, the leaves should be sorted such that corresponding leaves and leaf indices have the
+/// same position in both the `leaves` vector and the `leaf_indices` vector contained in the
+/// [primitives::BatchProof]
+pub fn verify_leaves_proof<H, L>(
 	root: H::Output,
-	leaf: Node<H, L>,
-	proof: primitives::Proof<H::Output>,
+	leaves: Vec<Node<H, L>>,
+	proof: primitives::BatchProof<H::Output>,
 ) -> Result<bool, Error>
 where
 	H: sp_runtime::traits::Hash,
 	L: primitives::FullLeaf,
 {
 	let size = NodesUtils::new(proof.leaf_count).size();
-	let leaf_position = mmr_lib::leaf_index_to_pos(proof.leaf_index);
+
+	let leaves_and_position_data = proof
```

Comment:
Same here: could you, please, add a check that number of entries in the `leaves` are matching number of entries in the `proof.leaf_indices`? Someone may call this function without checking this and it may return `true` even if proof not valid - e.g. there are entries in `leaves`, but there are no entries in `proof.leaf_indices` -> `leaves_positions_and_data` is empty -> function will return `true`